app/com/linkedin/drelephant/tuning/Flow.java

```diff
@@ -0,0 +1,71 @@
+package com.linkedin.drelephant.tuning;
+
+import com.linkedin.drelephant.tuning.Schduler.AzkabanJobStatusManager;
+import com.linkedin.drelephant.tuning.engine.MRExecutionEngine;
+import com.linkedin.drelephant.tuning.engine.SparkExecutionEngine;
+import com.linkedin.drelephant.tuning.hbt.AlgorithmManagerHBT;
+import com.linkedin.drelephant.tuning.hbt.BaselineManagerHBT;
+import com.linkedin.drelephant.tuning.hbt.FitnessManagerHBT;
+import com.linkedin.drelephant.tuning.obt.AlgorithmManagerOBT;
+import com.linkedin.drelephant.tuning.obt.BaselineManagerOBT;
+import com.linkedin.drelephant.tuning.obt.FitnessManagerOBT;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+
+public class Flow {
+  Map<String, List<Manager>> pipelines = null;
+
+  public Flow() {
+    pipelines = new HashMap<String, List<Manager>>();
+    createBaseLineManagersPipeline();
+    createJobStatusManagersPipeline();
+    createFitnessManagersPipeline();
+    createAlgorithmManagersPipeline();
+  }
+
+  public void executeFlow() {
+    for (final String pipelineType : this.pipelines.keySet()) {
+      new Thread(new Runnable() {
+        @Override
+        public void run() {
+          for (Manager manager : pipelines.get(pipelineType)) {
+            manager.execute();
+          }
+        }
+      }).start();
+    }
+  }
+
+  public void createBaseLineManagersPipeline() {
+    List<Manager> baselineManagers = new ArrayList<Manager>();
+    baselineManagers.add(new BaselineManagerHBT());
+    baselineManagers.add(new BaselineManagerOBT());
+    this.pipelines.put("baselineManager", baselineManagers);
+  }
+
+  public void createJobStatusManagersPipeline() {
+    List<Manager> jobStatusManagers = new ArrayList<Manager>();
+    jobStatusManagers.add(new AzkabanJobStatusManager());
+    //jobStatusManagers.add(new JobStatusManagerOBT());
+    this.pipelines.put("jobStatusManagers", jobStatusManagers);
+  }
+
+  public void createFitnessManagersPipeline() {
+    List<Manager> fitnessManagers = new ArrayList<Manager>();
+    fitnessManagers.add(new FitnessManagerHBT());
+    fitnessManagers.add(new FitnessManagerOBT());
+    this.pipelines.put("fitnessManagers", fitnessManagers);
+  }
+
+  public void createAlgorithmManagersPipeline() {
+    List<Manager> algorithmManagers = new ArrayList<Manager>();
+    algorithmManagers.add(new AlgorithmManagerHBT(new com.linkedin.drelephant.tuning.engine.MRExecutionEngine()));
+    algorithmManagers.add(new AlgorithmManagerHBT(new com.linkedin.drelephant.tuning.engine.SparkExecutionEngine()));
```

Comment:
How about having the execution engine i.e. Spark/MR being invoked inside AlgorithmManagerOBT/HBT? This can be tuning implementation specific. Having abstraction like this would ensure that we may not have to implement all execution types for a particular tuning type and we can probably not show it on the UI as well for a particular job type.